ast/ksp/src/main/kotlin/me/tatarka/kotlin/ast/KSAst.kt

```diff
@@ -465,9 +465,14 @@ private class KSAstType private constructor(
         get() = type.declaration.shortName
 
     override val arguments: List<AstType>
-        get() = type.arguments.map {
-            KSAstType(resolver, it.type!!)
-        }
+        get() = type.arguments
+            .mapNotNull {
+                // The type can be null for STAR use-site variance. In some cases KSP then cannot resolve the type.
+                it.type
+            }
```

Comment:
I worry that skipping over args would lead to invalid code-gen, maybe we should bail out in this case? Depends on where these arguments are being used I guess.

Reply:
I thought about other options, e.g. making the `type` nullable, but this would cause a lot of cascading changes in `KSAstType`. I thought about implementing `AstType` for star projections, but even then I'd need to throw an exception for certain APIs.

Reply:
I know this is nice to have, but I wonder of kotlin-inject has to do this:
```
        for (typeArg in type.arguments) {
            val argQualifier = typeArg.annotationAnnotatedWith(packageName, simpleName)
            if (argQualifier != null) {
                provider.error("Qualifier: $argQualifier can only be applied to the outer type", typeArg)
            }
            checkTypeArgs(packageName, simpleName, typeArg)
        }
```
Without this loop the problem is gone, too.

Reply:
The important thing is I'd want to it crash in other places where we'd be handling this incorrectly instead of skipping over the argument and generating invalid code. Did you mean you tried to changing it to `override val arguments: List<AstType?>`? If you don't want to go an inspect every single usage now then putting `!!` should be ok, it'll crash at that site in the future and we can take it from there. Could even add something like

```
fun AstType.expectArguments(): List<AstType>
```
to make it more clear that that is what we are doing.

> Right now it's crashing when checking for qualifiers.

For that specific case, yeah I think skipping it would be acceptable. That code will need to be refactored anyway for https://github.com/evant/kotlin-inject/issues/409 fyi though I guess that's neither here nor there.

Reply:
> Without this loop the problem is gone, too.

The check was added to punt on more complex qualifier handling, right now only a single qualifier on a type is supported.

Reply:
New revision. Let me know what you think. I kept the crashing behavior for now and handle the exception in this one particular case.